### PR TITLE
implement credential management

### PR DESCRIPTION
**Pull request by thommay**
per RFC 99

### CODE REVIEW

File: chef-config/lib/chef-config/workstation_config_loader.rb [hunk 1]
```diff
@@ -138,6 +146,34 @@ def working_directory
       a
     end
 
+    def apply_credentials(creds, _profile)
+      Config.node_name = creds.fetch("node_name") if creds.key?("node_name")
+      Config.node_name = creds.fetch("client_name") if creds.key?("client_name")
```
Comment (adamleff) [inline chef-config/lib/chef-config/workstation_config_loader.rb]: I wonder if this is worthy of a warning log message or similar if the user has both a `node_name` and a `client_name` in their credentials.

File: chef-config/lib/chef-config/workstation_config_loader.rb [hunk 2]
```diff
@@ -138,6 +146,34 @@ def working_directory
       a
     end
 
+    def apply_credentials(creds, _profile)
```
Comment (adamleff) [inline chef-config/lib/chef-config/workstation_config_loader.rb]: Is accepting the profile here just future-proofing? Or am I missing some context (totally reasonable assumption on your part).

File: chef-config/lib/chef-config/workstation_config_loader.rb (hunk 1 above)
Comment (thommay) [reply chef-config/lib/chef-config/workstation_config_loader.rb]: Yeah that really needs a raise.

**Comment by lamont-granquist** [comment]
still 👍

**Pull request merged**